From: Dimitri John Ledkov <xnox@ubuntu.com>
Date: Tue, 25 Jul 2017 12:21:23 +0100
Subject: Ship modprobe.d drop-in to set bonding max_bonds to 0

This allows networkd to correctly manage bond0 using networkd, when requested
by the user.

Fixes #5971 #6184

(cherry picked from commit b272ba9230db94b7db73271c56e0b6771e2281b6)
---
 meson.build             |  4 ++++
 modprobe.d/systemd.conf | 14 ++++++++++++++
 2 files changed, 18 insertions(+)
 create mode 100644 modprobe.d/systemd.conf

diff --git a/meson.build b/meson.build
index eabf215..00ef5c3 100644
--- a/meson.build
+++ b/meson.build
@@ -73,6 +73,7 @@ polkitpkladir = join_paths(localstatedir, 'lib/polkit-1/localauthority/10-vendor
 varlogdir = join_paths(localstatedir, 'log')
 xinitrcdir = join_paths(sysconfdir, 'X11/xinit/xinitrc.d')
 rpmmacrosdir = get_option('rpmmacrosdir')
+modprobedir = join_paths(rootprefixdir, 'lib/modprobe.d')
 
 # Our own paths
 pkgdatadir = join_paths(datadir, 'systemd')
@@ -2303,6 +2304,8 @@ install_data('xorg/50-systemd-user.sh',
              install_dir : xinitrcdir)
 install_data('system-preset/90-systemd.preset',
              install_dir : systempresetdir)
+install_data('modprobe.d/systemd.conf',
+             install_dir : modprobedir)
 install_data('README',
              'NEWS',
              'CODING_STYLE',
@@ -2389,6 +2392,7 @@ status = [
         'PAM modules dir:                   @0@'.format(pamlibdir),
         'PAM configuration dir:             @0@'.format(pamconfdir),
         'RPM macros dir:                    @0@'.format(rpmmacrosdir),
+        'modprobe.d dir:                    @0@'.format(modprobedir),
         'D-Bus policy dir:                  @0@'.format(dbuspolicydir),
         'D-Bus session dir:                 @0@'.format(dbussessionservicedir),
         'D-Bus system dir:                  @0@'.format(dbussystemservicedir),
diff --git a/modprobe.d/systemd.conf b/modprobe.d/systemd.conf
new file mode 100644
index 0000000..55a9a7d
--- /dev/null
+++ b/modprobe.d/systemd.conf
@@ -0,0 +1,14 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+#
+# When bonding module is loaded, it creates bond0 by default due to max_bonds
+# option default value 1. This interferes with the network configuration
+# management / networkd, as it is not possible to detect whether this bond0 was
+# intentially configured by the user, or should be managed by
+# networkd/NM/etc. Therefore disable bond0 creation.
+
+options bonding max_bonds=0
